tests/unit/geo.tcl

```diff
@@ -456,4 +579,74 @@ start_server {tags {"geo"}} {
+            r del mypoints
+
+            geo_random_point search_lon search_lat
+            set width_m [expr {[randomInt 10000]+10}]
+            set height_m [expr {[randomInt 10000]+10}]
+            set lat_delta [geo_raddeg [expr {$height_m/2/6372797.560856}]]
+            set long_delta_top [geo_raddeg [expr {$width_m/2/6372797.560856/cos([geo_degrad [expr {$search_lat+$lat_delta}]])}]]
+            set long_delta_middle [geo_raddeg [expr {$width_m/2/6372797.560856/cos([geo_degrad $search_lat])}]]
+            set long_delta_bottom [geo_raddeg [expr {$width_m/2/6372797.560856/cos([geo_degrad [expr {$search_lat-$lat_delta}]])}]]
+
+            # Total of 8 points are generated, which are located at each vertex and the center of each side
+            set bounds(0) [expr {$search_lon-$long_delta_top}]
+            set bounds(1) [expr {$search_lat+$lat_delta}]
+            set bounds(2) [expr {$search_lon+$long_delta_top}]
+            set bounds(3) [expr {$search_lat+$lat_delta}]
+            set bounds(4) [expr {$search_lon+$long_delta_bottom}]
+            set bounds(5) [expr {$search_lat-$lat_delta}]
+            set bounds(6) [expr {$search_lon-$long_delta_bottom}]
+            set bounds(7) [expr {$search_lat-$lat_delta}]
+            set bounds(8) $search_lon
+            set bounds(9) [expr {$search_lat+$lat_delta}]
+            set bounds(10) $search_lon
+            set bounds(11) [expr {$search_lat-$lat_delta}]
+            set bounds(12) [expr {$search_lon+$long_delta_middle}]
+            set bounds(13) $search_lat
+            set bounds(14) [expr {$search_lon-$long_delta_middle}]
+            set bounds(15) $search_lat
+
+            lappend debuginfo "Search area: geosearch mypoints fromlonlat $search_lon $search_lat bybox $width_m $height_m m"
+            set tcl_result {}
+            for {set i 0} {$i < 16} {incr i 2} {
+                set x $bounds($i)
+                set y $bounds([expr {$i+1}])
+                r geoadd mypoints $x $y place:$i
+                lappend tcl_result "place:$i"
+                lappend debuginfo "geoadd mypoints $x $y place:$i"
+            }
+
+            set res2 [lsort $tcl_result]
+
+            # move the box by two meter to put the coordinate slightly inside the box.
+            set height_new [expr {$height_m+4}]
+            set width_new [expr {$width_m+4}]
+            set res [lsort [r geosearch mypoints fromlonlat $search_lon $search_lat bybox $width_new $height_new m]]
+            if {$res != $res2} {
+                set diff [compare_lists $res $res2]
+                lappend debuginfo "diff: $diff"
+                fail "place should be found, debuginfo: $debuginfo, height_new: $height_new width_new: $width_new"
+            }
+
+            # move the box by two meter to put the coordinate slightly outside the box.
+            set height_new [expr {$height_m-4}]
+            set width_new [expr {$width_m-4}]
```

Comment:
In fact, we currently reduce both height and width, which can cover the situation you mentioned (only reduce height or width), but I added the test case you mentioned.

Reply:
the reason i didn't like the previous test was that if we have a bug in the corners handing, and that bug only affects the longitude, the test wouldn't have found the bug.
i.e. because reducing the height would have moved the point outside the search area, and we wouldn't have known that reducing the width would have still found it.